elasticsearch_collectd.py

```diff
@@ -900,9 +908,16 @@ def fetch_stats(self):
 
         node_json_stats = self.fetch_url(self.es_node_url)
         if node_json_stats:
-            if self.es_cluster is None:
-                self.es_cluster = node_json_stats['cluster_name']
+            # Only if Cluster name is not provided as a config option, use the
+            # value retured from the ES endpoint
+            if self.es_cluster_from_config is None:
+                es_cluster_from_endpoint = node_json_stats['cluster_name']
+                # Check if cluster name matches from what the plugin saw in the
+                # previous interval, in not, update the cluster name
+                if es_cluster_from_endpoint != self.es_cluster:
+                    self.es_cluster = es_cluster_from_endpoint
             else:
+                self.es_cluster = self.es_cluster_from_config
                 log.info('Configured with cluster_json_stats=%s' %
```

Comment:
I would move this log statement outside the else statement so it always gets shown.